core/vm/eips.go

```diff
@@ -174,3 +175,13 @@ func opBaseFee(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]
 	scope.Stack.push(baseFee)
 	return nil, nil
 }
+
+func enable3860(jt *JumpTable) {
+	createOp := *jt[CREATE]
```

Comment:
I clone `operation` struct here, because doing just `jt[CREATE].dynamicCost = ...` would modify existing operation in underlying global `londonInstructionSet`, and that would influence unit tests that switch between enabling / disabling EIP3860.

Not sure if this is also a problem for other `enable...` functions above (`enable2200`, `enable3529`)

Reply:
Shouldn't be an issue -- NewXXInstructionSet uses a brand new instructionset.

Reply:
But `newLondonInstructionSet()` is called only once when global is initialized https://github.com/ethereum/go-ethereum/blob/17f1c2dc0f070599db6a792eba63c51ea93e17c8/core/vm/jump_table.go#L60

`NewEVMInterpreter` then creates a new `JumpTable` instance, but it contains the pointers to the same opertations of `londonInstructionSet` global.
https://github.com/ethereum/go-ethereum/blob/17f1c2dc0f070599db6a792eba63c51ea93e17c8/core/vm/interpreter.go#L78

Reply:
Better fix for this is in https://github.com/ethereum/go-ethereum/pull/23977

Reply:
Does this code needs any changes after #23977?

Reply:
> Better fix for this is in #23977

Actually this hack is still needed, because `JumpTable` contains pointers to `operation`s, so modification like `jt[CREATE].dynamicGas = gasCreateEip3860` would change the operation for `londonInstructionSet`, too.

Reply:
This is now not needed, I think, we changed it so ops are deep-copied.

Reply:
In the latest commit I removed it anyway, because the new initcode charge is applied inside create implementation and not in dynamicGas calculation, so new cost functions are not needed.